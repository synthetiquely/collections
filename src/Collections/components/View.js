import React, { Component } from 'react';
import glamorous from 'glamorous';

// const Container = glamorous.div(
//   {
//     // display: 'flex',
//     // flexDirection: 'column',
//     // justifyContent: 'center',
//     // alignItems: 'center',
//     ':hover': {
//       boxShadow: '1px 1px 1px #ccc',
//     },
//     cursor: 'pointer',
//   },
//   ({ size }) => {
//     if (size === 'large') {
//       return {
//         gridColumn: 'span 3',
//         gridRow: 'span 1',
//       };
//     } else if (size === 'medium') {
//       return {
//         gridColumn: 'span 2',
//         gridRow: 'span 1',
//       };
//     }
//     return {
//       gridColumn: 'span 1',
//       gridRow: 'span 1',
//     };
//   },
// );

const Image = glamorous.img(
  {
    minWidth: '100%',
    maxHeight: '150px',
    objectFit: 'cover',
  },
  // ({ imageLoaded }) => {
  //   if (!imageLoaded) {
  //     return {
  //       backgroundRepeat: 'no-repeat',
  //       backgroundImage: 'linear-gradient(to right, #ffdb4d, #ffe78c)',
  //       backgroundSize: '100%, 100%',
  //       backgroundPosition: '0 0',
  //     };
  //   }
  //   return {};
  // },
  ({ size }) => {
    if (size === 'large') {
      return {
        gridColumn: 'span 3',
        gridRow: 'span 1',
      };
    } else if (size === 'medium') {
      return {
        gridColumn: 'span 2',
        gridRow: 'span 1',
      };
    }
    return {
      gridColumn: 'span 1',
      gridRow: 'span 1',
    };
  },
);
export default class View extends Component {
  constructor(props) {
    super(props);
    this.state = {
      imageLoaded: false,
    };
  }

  onLoad(status) {
    this.setState({
      imageLoaded: status,
    });
  }

  render() {
    const {
      title, src, size, onClick,
    } = this.props;

    return (
      <Image
        size={size}
        src={src}
        alt={title}
        onClick={onClick}
        onLoad={() => this.onLoad(true)}
        imageLoaded={this.state.imageLoaded}
      />
    );
  }
}
