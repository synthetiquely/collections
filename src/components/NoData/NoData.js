import glamorous from 'glamorous';

const NoData = glamorous.div({
  maxWidth: '300px',
  margin: '0 auto',
  padding: '10px',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  textAlign: 'center',
  color: '#a9a9a9',
  fontSize: '1.1rem',
});

export default NoData;
