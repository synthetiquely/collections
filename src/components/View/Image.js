import glamorous from 'glamorous';

const Image = glamorous.img(
  {
    opacity: '0',
    width: '100%',
    verticalAlign: 'bottom',
    willChange: 'opacity',
    transition: 'opacity 2s',
  },
  ({ loaded }) => {
    if (loaded) {
      return {
        opacity: '1',
      };
    }
    return {};
  },
);

export default Image;
