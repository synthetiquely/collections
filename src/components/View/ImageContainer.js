import glamorous from 'glamorous';
import { SIZE_RATIO } from '../../constants';

const ImageContainer = glamorous.div(
  {
    margin: '2.5px',
    cursor: 'pointer',
    willChange: 'width, flexGrow',
    ':hover': {
      boxShadow: '1px 1px 1px #ccc',
    },
  },
  ({ height, width }) => {
    if (height && width) {
      const size = Math.floor(width * SIZE_RATIO / height);

      return {
        flexGrow: `${size}`,
        width: `${size}px`,
      };
    }
    return {};
  },
);

export default ImageContainer;
