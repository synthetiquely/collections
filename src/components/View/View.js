import React, { Component } from 'react';
import glamorous from 'glamorous';
import { SIZE_RATIO } from '../../constants';

const ImageContainer = glamorous.div(
  {
    margin: '2.5px',
    cursor: 'pointer',
    willChange: 'width, flexGrow',
    ':hover': {
      boxShadow: '1px 1px 1px #ccc',
    },
  },
  ({ height, width }) => {
    if (height && width) {
      const size = Math.floor(width * SIZE_RATIO / height);

      return {
        flexGrow: `${size}`,
        width: `${size}px`,
      };
    }
    return {};
  },
);

const Image = glamorous.img(
  {
    opacity: '0',
    width: '100%',
    verticalAlign: 'bottom',
    willChange: 'opacity',
    transition: 'opacity 2s',
  },
  ({ loaded }) => {
    if (loaded) {
      return {
        opacity: '1',
      };
    }
    return {};
  },
);

export default class View extends Component {
  constructor(props) {
    super(props);
    this.state = { loaded: false };
    this.width = 0;
    this.height = 0;
    this.img = null;
    this.onLoad = this.onLoad.bind(this);
    this.onError = this.onError.bind(this);
    this.setRef = this.setRef.bind(this);
  }

  onLoad() {
    /**
     * Due to the fact that img width and height will be known in advance
     * as soon as we bring API to our app,
     * we could remove this lines of code.
     */
    this.width = this.img.naturalWidth || this.img.width;
    this.height = this.img.naturalHeight || this.img.height;
    this.setState({
      loaded: true,
    });
  }

  // @TODO: handle error case
  onError() {}

  setRef(img) {
    this.img = img;
  }

  render() {
    const { src, title, onClick } = this.props;

    return (
      <ImageContainer onClick={onClick} width={this.width} height={this.height}>
        <Image
          src={src}
          alt={title}
          innerRef={this.setRef}
          loaded={this.state.loaded}
          onLoad={this.onLoad}
          onError={this.onError}
        />
      </ImageContainer>
    );
  }
}
