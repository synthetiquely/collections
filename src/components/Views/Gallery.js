import glamorous from 'glamorous';

const Gallery = glamorous.section(
  {
    display: 'flex',
    flexWrap: 'wrap',
    '@media only screen and (min-width: 768px)': {
      '&::after': {
        content: "''",
        flexGrow: '999999999',
      },
    },
  },
  ({ isOpen }) => ({
    overflowY: isOpen ? 'hidden' : 'auto',
  }),
);

export default Gallery;
