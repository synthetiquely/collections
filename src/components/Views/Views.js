import React from 'react';
import View from '../View/View';
import Gallery from './Gallery';

const Views = ({ images, isOpen, onClick }) => {
  if (!images) {
    return null;
  }

  const renderItems = () =>
    images.map((item, index) => (
      <View
        src={item.src}
        title={item.title}
        key={item.id}
        onClick={onClick(index)}
      />
    ));

  return <Gallery isOpen={isOpen}>{renderItems()}</Gallery>;
};

export default Views;
