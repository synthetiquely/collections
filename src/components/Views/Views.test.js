import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Views from './Views';
import NoData from '../NoData/NoData';
import View from '../View/View';

describe('Views Component', () => {
  const props = {
    images: [
      {
        id: '1',
        height: 300,
        width: 250,
        color: 'red',
        src:
          'https://n6-img-fp.akamaized.net/free-vector/abstract-gray-background-with-lines_1108-73.jpg?size=338&ext=jpg',
        description: 'Test Image 1',
      },
      {
        id: '2',
        height: 333,
        width: 253,
        color: 'red',
        src:
          'https://n6-img-fp.akamaized.net/free-vector/abstract-gray-background-with-lines_1108-73.jpg?size=338&ext=jpg',
        description: 'Test Image 2',
      },
      {
        id: '3',
        height: 444,
        width: 254,
        color: 'red',
        src:
          'https://n6-img-fp.akamaized.net/free-vector/abstract-gray-background-with-lines_1108-73.jpg?size=338&ext=jpg',
        description: 'Test Image 3',
      },
    ],
    onClick: jest.fn(),
    loadItems: jest.fn(),
    isLoading: false,
    isOpen: false,
  };
  it('should render and match its own snapshot', () => {
    const wrapper = mount(<Views {...props} />);
    const tree = renderer.create(<Views {...props} />).toJSON();

    expect(wrapper).toHaveLength(1);
    expect(tree).toMatchSnapshot();
  });

  it('should render No Data component if there are no images', () => {
    const testProps = {
      ...props,
      images: [],
    };
    const wrapper = mount(<Views {...testProps} />);

    expect(wrapper.find(NoData)).toHaveLength(1);
  });

  it('should render view components', () => {
    const wrapper = mount(<Views {...props} />);

    expect(wrapper.find(View)).toHaveLength(props.images.length);
  });
});
