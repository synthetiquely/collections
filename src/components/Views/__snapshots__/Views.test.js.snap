// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views Component should render and match its own snapshot 1`] = `
<div>
  <section
    className="css-wcs7yy"
  >
    <div
      className="css-1ykgg8r"
      onClick={undefined}
      tabIndex={0}
    >
      <img
        alt="Test Image 1"
        className="css-eob8a0"
        onError={[Function]}
        onLoad={[Function]}
        src="https://n6-img-fp.akamaized.net/free-vector/abstract-gray-background-with-lines_1108-73.jpg?size=338&ext=jpg"
      />
    </div>
    <div
      className="css-rgncr2"
      onClick={undefined}
      tabIndex={0}
    >
      <img
        alt="Test Image 2"
        className="css-eob8a0"
        onError={[Function]}
        onLoad={[Function]}
        src="https://n6-img-fp.akamaized.net/free-vector/abstract-gray-background-with-lines_1108-73.jpg?size=338&ext=jpg"
      />
    </div>
    <div
      className="css-1phrjfd"
      onClick={undefined}
      tabIndex={0}
    >
      <img
        alt="Test Image 3"
        className="css-eob8a0"
        onError={[Function]}
        onLoad={[Function]}
        src="https://n6-img-fp.akamaized.net/free-vector/abstract-gray-background-with-lines_1108-73.jpg?size=338&ext=jpg"
      />
    </div>
  </section>
</div>
`;
